src/crimson/net/SocketConnection.cc

```diff
@@ -893,32 +893,31 @@ SocketConnection::execute_open()
 {
   logger().debug("{} trigger open, was {}", *this, static_cast<int>(state));
   state = state_t::open;
+  // satisfy the handshake's promise
+  h.promise.set_value();
```

Comment:
i'd suggest move this back to `start_connect()` and `start_accept()`. where we also do `h.promise.set_value()` in the exception handling.

Reply:
https://github.com/ceph/ceph/pull/25716#discussion_r245226646
Agree to move back, but I think this is better to be with `execute_open()` as atomic operations.

Reply:
personally i still want them to be located in the same function. but yeah, your rationale makes sense too!